akka-stream/src/main/scala/akka/stream/scaladsl/Sink.scala

```diff
@@ -220,4 +220,11 @@ object Sink extends SinkApply {
     new Sink(new AcknowledgeSink(bufferSize, DefaultAttributes.acknowledgeSink, shape("AcknowledgeSink"), timeout))
   }
 
+  /**
+   * Crates a `Sink` that is materialized into `Source` to connect with another flow.
+   * This sink gets backpressure from second flow and does not have internal buffer.
+   */
+  def pipedSink[T](): Sink[T, Source[T, Unit]] =
+    Sink.publisher[T].mapMaterializedValue(Source(_)).named("PipedSink")
```

Comment:
In fact, we had that exact philosophy on Play - APIs should return iteratees, not accept enumerators.  It's great in theory, in practice, it breaks down when you need to do simple plumbing - and it turns out that when dealing with IO, most people just want to do simple plumbing.  So for a number of releases, due to our ideals, we were making life difficult for most of our users.  In the end we introduced new APIs that broke these ideals, but made these common plumbing use cases very straight forward, and an interesting thing about that is that since then, there have been much fewer questions about iteratees/enumerators on the mailing list, we made things that should have been easy, easy, and people found it easy.